lisa/tests/scheduler/load_tracking.py

```diff
@@ -971,7 +979,44 @@ def get_trace_cpu_util(self):
             phase_df = df_window(df, (start, end), method='pre', clip_window=True)
 
             for cpu in self.cpus:
-                util = phase_df[phase_df['cpu'] == cpu]['util']
+
+                cpu_phase_df = phase_df[phase_df['cpu'] == cpu].copy()
+
+                for task in tasks:
+                    if self.rtapp_profile[task].phases[phase-1].cpus.count(cpu):
```

Comment:
Should I raise a runtime error in case phase = row.phase is not greater than 1?

```
         cpu_util = {}
         for row in self.trace.analysis.rta.df_phases(migr_task).itertuples():
             phase = row.phase
+
+            if phase < 1:
+                raise RuntimeError('phase has to be greater than 0')
+
             duration = row.duration
             start = row.Index
             end = start + duration
```

Reply:
Why do we need to look at the previous phase rather than the current one ?

Reply:
phase = row.phase gives me 1 or 2

wlgen_task.phases is a list with two members:

(Pdb) wlgen_task.phases[0].cpus
[0]
(Pdb) wlgen_task.phases[1].cpus
[3]
(Pdb) wlgen_task.phases[2].cpus
*** IndexError: list index out of range

So it's just to handle this transition from {1,2} to {0,1}

Reply:
ah right, phases are not named which leads to problems with the buffer phase. There has been plans for more than a year to rework wlgen to fix that but lack of manpower on that front has delayed it.

For now, just skip when `phase == 0` with that comment to make it easy to remove the hack when wlgen is fixed. Looking at the CPU of the previous phase is not what we want to do:
```
            # TODO: remove that once we have named phases to skip the buffer phase
```